### ISSUE TITLE
Add thread_id to logs

### ISSUE THREAD

**Issue by breezewish** [tikv/tikv#13395]
## Enhancement

It is possible to add a field to every log like like `[tid=xxx]` or `[thread_id=xxx]` by using [`thread::current().id()`](https://doc.rust-lang.org/std/thread/struct.ThreadId.html). Although this Thread ID cannot be mapped to the Linux LWP, it can be used to learn the processing context of a worker thread, e.g. what and when of previous operations. This could be useful to troubleshoot latency problems.

A working implementation could be patching slog like https://github.com/breezewish/slog/commit/e29b09a2f53282b85ec5c259238cb739a5b44a60 . It may be also possible to use the [FnValue as suggested in the discussion](https://github.com/slog-rs/slog/discussions/313).

**Comment by BusyJay** [comment]
Using tid instead of `thread::current().id()` is better as we can map it across language and runtime.

**Comment by 2022H1030014G** [comment]
Hi @breezewish I hope you are doing great, can you assign this issue to me.

**Comment by TheSwarnim** [comment]
Hi @BusyJay, @breezewish. I'm new here and wanted to pick this task. I find out that there is no update on this task for more than a month. Can you assign this task to me please?

**Comment by BusyJay** [comment]
@2022H1030014G do you still want to pick up the issue?

**Comment by breezewish** [comment]
I see that @2022H1030014G has claimed this previously, may be we can have some discussions about whether there are some updates.

**Comment by BusyJay** [comment]
A week has passed. I think @TheSwarnim can start the work (if you are still interested).

**Comment by callmetushar123** [comment]
is it up for grabs?

**Comment by jgrande** [comment]
Hi! If this is still up for grabs, I would like to contribute a patch.

**Comment by jgrande** [comment]
@BusyJay @breezewish I know this is quite old, but are you still interested in this? I'm getting started with TiKV and Rust and would be happy to contribute a patch.

**Comment by BusyJay** [comment]
@breezewish @tonyxuqqi PTAL

**Comment by breezewish** [comment]
@jgrande Yes, looks to be a good idea!

**Comment by YuJuncen** [comment]
I have noticed a problem. For now we indeed added a `thread_id` field to each log. But it seems all logs' `thread_id` is the same. I guess this is probably because the `Async` drain evaluates the log entries in a background thread (namely `slog` in TiKV.), so all logs' `thread_id` point to it.

TBH I have no idea about how to solve this.😢 It seems we cannot trivially append a entry which evaluates at the callsite of `log!` macro. But I have noticed the log macros are provided in `tikv_util`, I guess changing those macros might be helpful.

**Comment by jgrande** [comment]
Thanks for pointing this out @YuJuncen. You are right, the problem happens when using the async logger. I prototyped modifying the macros to fetch the thread id there, however one problem I found is that many log messages are not going through the macros.

Until we find a good solution, I think I should revert the change. Alternatively, I can scope it to the sync logger, but I'm not sure how much value that will have.

@tonyxuqqi @Connor1996 thoughts? Tagging you as you reviewed the PR. Thanks!
